### DISCUSSION TITLE
Relative "twos' complement" works a bit wonky, doesn't it? Please help understand, maybe it's a bug?

### DISCUSSION THREAD

**Discussion by katerlouis** [rsjaffe/MIDI2LR#684] [Problems]
I want to use my OP-1 Field (pocket synthesizer) via bluetooth midi as a controller for Lightroom. The knobs can be configured inside the device to act relatively. Turn knob clockwise: 1 is sent; Turn knob counterclockwise: 127 is sent. This is defined [here](https://github.com/rsjaffe/MIDI2LR/wiki/MIDI-Controller-Setup#controllers-that-send-relative-values-or-have-unusual-ranges) as "two's complement".

Inside MIDI2LR "Basic Tone -> Exposure" configured as "two's complement" works pretty good already. But I have two issues:

1) Whenever I change the turning direction, the first "new direction message" is not affecting the slider inside Lightroom. 
Example: Exposure is at 0,00. I turn the knob clockwise 5 clicks at a resolution of 250 resulting in an exposure of +0,36. Without a pause, I turn the knob counterclockwise (change in direction) for 1 click: Nothing happens. The CC message has been registered  by both MIDI Monitor and MIDI2LR but the slider hasn't moved. In fact: When I go back and forth, the messages all get picked up, but the slider still doesn't move. When I wait between direction changes for roughly 3+ seconds, the new direction click does change the slider. This might sound minor, but when adjusting to taste I constantly go back and forth and "a dead click" like that doesn't feel nice. I have no explanation to why that happens, because quick consecutive clicks in the same direction work like a charm.

2) The in- and decrease in value per click is not really constant and feels really wanky. Example: Exposure at 0,00 with 8 clicks clockwise at a resolution of 250 always results in +0,60. What is odd is: The "first click in a chain" always adds (or subtracts) 0,04 while all consecutive clicks add (or subtract) 0,08. When I wait wait the beforementioned 3+ seconds, apparently "a new chain starts". 

This seems to have something to do with issue 1 aswell. Example on resolution 250: 
* click 1 (clockwise): exposure is at +0,04
* click 2 (clockwise): exposure is at +0,12
* click 3 (clockwise): exposure is at +0,20
* click 4 (counterclockwise): exposure remains +0,20
* click 5 (counterclockwise): exposure is at +0,12
* click 6 (counterclockwise): exposure is at +0,04
* click 7 (counterclockwise): exposure is at -0,04 (!)

To sum up:
* The first click is only half the step size than consecutive clicks
* A turn around click does nothing

MIDI Monitor doesn't hint to any difference in the first click or the turn around click. The data looks exactly the same to me (MIDI novice)

I tried playing around with different resolutions and couldn't find anything working differently.

Questions:
a) Is this behavior intended (If so: Why?)
b) Can this be set up to work just like the UP and DOWN arrow keys? 1 click = +/-0,10 in exposure without the quirkyness?

Thanks!
R. 

[EDIT] I just found out: When you go back and forth and back and forth and so on, at some point roughly after the 3+ seconds a new "first message" seems to be applied to the slider with +/-0,04, obviously depending on the direction that just happened to be triggered right after that magic timeout. This timeout also seems to be constant.

**Comment by katerlouis** [comment]
@rsjaffe gentle push :)

**Comment by PitToad** [comment]
Announcement from June 16;

"I will be busy and unable to do much with this project until at least the beginning of September 22. In the meantime, feel free to help each other with issues. If you are a programmer, please feel free to fork this project and work on any issues or improvements.

Thank you. Rory."

**Comment by katerlouis** [comment]
@rsjaffe September 22 has passed, so here's another friendly push 😊 – If you need any assistance in replicating this issue or have something I could test to further track down the problem, please let me know!
